Remove transitive interfaces (implicitly inherited ones)

clue opened the pull request:
Before:
![uml-old](https://f.cloud.github.com/assets/776829/1820564/3acf0174-70f3-11e3-8d50-1b29ebffaada.png)

After:
![uml-new](https://f.cloud.github.com/assets/776829/1820563/3aa83bca-70f3-11e3-8173-f1e64d584031.png)

Fixes #10.
Not sure how this relates to #17.

clue committed:
Remove transitive interfaces (implicitly inherited ones)

clemens-tolboom commented:
The result looks awesome :)

But this needs a test somewhat in line with #17 (in which I failed to write a proper one :()

clemens-tolboom commented on src/Fhaculty/Graph/Uml/ClassDiagramBuilder.php:
```diff
@@ -543,4 +543,27 @@ private function escape($id)
+        // remove each interface already implemented by any of the inherited interfaces
+        foreach ($interfaces as $if) {
+            foreach ($if->getInterfaceNames() as $in) {
+                //unset($interfaces[$in]);
```
This is not OK ... this commented unset makes the loops unless.

Should we remove all?

clue replied:
Duh, did I actually just comment out _the one line_ that actually fixes this? :>

I've pushed an update :D

Yeah, this was a last-minute change in order to produce the "old" graph. Looks like we need some tests :)

clemens-tolboom committed:
Fixed merge conflict.

clue committed:
Don't just comment out the fix :>
Don't just comment out the fix :>
Don't just comment out the fix :>
Don't just comment out the fix :>
Don't just comment out the fix :>

clemens-tolboom committed:
Merge branch 'fix-transitive-interfaces' of github.com:clue/graph-uml into fix-transitive-interfaces

clemens-tolboom commented:
Lol ... working both with the speed of light.

clue commented:
Yeah, what a coincidence :)

Feel free to merge this if it's reviewed.

clemens-tolboom commented:
Awesome fix :)
